src/gaxios.ts

```diff
@@ -147,7 +147,7 @@ export class Gaxios {
 
     // baseUrl has been deprecated, remove in 2.0
     const baseUrl = opts.baseUrl || opts.baseURL;
-    if (baseUrl) {
+    if (baseUrl && !opts.url.startsWith(baseUrl)) {
```

Comment:
If we go with the approach of simply checking for `baseUrl` as a prefix, perhaps we could perform an additional check to ensure that we're not re-appending a scheme:

we never want:

```
https://www.baseurl.com/foo/http://www.oldurl.com
```

Perhaps we could do:

```js
const {parse, resolve} = require("url");
if (baseUrl && !opts.url.startsWith(baseUrl)) {
  opts.url  = resolve(baseUrl, parse(opts.url).path)
}
```